### PR TITLE
Revert amp-story-access validation.

### PR DESCRIPTION
**Pull request by gmajoulet**
Revert `amp-access` and `amp-story-access` validation within AMP Stories. It was never documented or communicated on.
Recipe 534681 by @newmuis should tell us more about its usage before we merge this PR.

#12180

### CODE REVIEW

**Comment by honeybadgerdontcare** [comment]
Please follow the process of Intent-to-Remove (I2R) documented under I2D: https://github.com/ampproject/amphtml/blob/6d103b5603feab284d247bca132dc08064659b14/spec/amp-versioning-policy.md#deprecations

Have you verified that no site is using this?

File: extensions/amp-story/validator-amp-story.protoascii [hunk 1]
```diff
@@ -104,7 +104,6 @@ tags: {  # <amp-story>
     child_tag_name_oneof: "AMP-GEO"
     child_tag_name_oneof: "AMP-PIXEL"
     child_tag_name_oneof: "AMP-SIDEBAR"
-    child_tag_name_oneof: "AMP-STORY-ACCESS"
```
Comment (honeybadgerdontcare) [inline extensions/amp-story/validator-amp-story.protoascii]: Blocking for I2D/I2R.

**Comment by gmajoulet** [comment]
As long as no story is using this (will be confirmed by the recipe mentioned in the first comment), it should be safe to merge this without following the I2D/I2R since it hasn't been documented or announced to anyone.
Let's wait for the results.

**Comment by honeybadgerdontcare** [comment]
@gmajoulet What recipe are you using? Are you sure that recipe covers all sites? Whenever we become more restrictive we must be sure about this which is why we created the I2D/I2R because we've had issues in the past.

**Comment by gmajoulet** [comment]
Recipe `534681` shows 0 URL using `amp-story-access`, which we expected as we didn't announce or document the feature in any way. Please let me know if you want us to still go through the I2D/I2R process. We thought it might introduce confusion as it's never been announced, but are happy to follow the usual process if you think it's worth it.

**Comment by honeybadgerdontcare** [comment]
I think you're right that it's not going to affect publishers. If you don't mind I'll run a sanity check to the recipe on a 10k set tonight.

**Comment by gmajoulet** [comment]
Sounds good, thank you for your help!

File: extensions/amp-story/validator-amp-story.protoascii (hunk 1 above)
Comment (honeybadgerdontcare) [reply extensions/amp-story/validator-amp-story.protoascii]: Unblocking. This does not affect any publishers.

For Googlers, if you're interested in running such a test please see http://go/amp-rule-change

**Comment by gmajoulet** [comment]
Re-requested a review, it was still marked as "changes requested" :)

**Pull request merged**